pincer/middleware/interaction_create.py

```diff
@@ -56,23 +64,59 @@ async def interaction_create_middleware(self, payload: GatewayDispatch):
         if should_pass_cls(command.call):
             kwargs["self"] = self
 
-        message = await command.call(**kwargs)
+        if isasyncgenfunction(command.call):
+            message = command.call(**kwargs)
+            started = False
 
-        if isinstance(message, Embed):
-            message = Message(embeds=[message])
-        elif not isinstance(message, Message):
-            message = Message(message) if message else Message(
-                self.received_message,
-                flags=InteractionFlags.EPHEMERAL
-            )
+            async for msg in message:
 
-        await self.http.post(
-            f"interactions/{interaction.id}/{interaction.token}/callback",
-            message.to_dict()
-        )
+                msg = convert_message(self, msg)
+                if started:
+                    try:
+                        await self.http.post(
+                            f"webhooks/{interaction.application_id}/{interaction.token}",
+                            msg.to_dict().get("data")
+                        )
+                    except RateLimitError as e:
+                        _log.exception(
+                            f"RateLimitError: {e}. Retrying in {e.json.get('retry_after', 40)} seconds")
+                        await sleep(e.json.get("retry_after", 40))
+                        await self.http.post(
+                            f"webhooks/{interaction.application_id}/{interaction.token}",
+                            msg.to_dict().get("data")
+                        )
+
+                else:
+                    started = True
+
+                    await self.http.post(
+                        f"interactions/{interaction.id}/{interaction.token}/callback",
+                        msg.to_dict()
+                    )
+                await sleep(0.3)
+        else:
+            message = await command.call(**kwargs)
+            message = convert_message(self, message)
+
+            await self.http.post(
+                f"interactions/{interaction.id}/{interaction.token}/callback",
+                message.to_dict()
+            )
 
     return "on_interaction_create", [interaction]
 
 
+def convert_message(self, message: Union[Embed, Message, str]) -> Message:
```

Comment:
Extraction this to a function is a great idea. Tho, when we'll have more and more thing to convert to message (button, list, ...) we might want to use a dictionary and bind type key to convert function or something like?